helpers/multi.d.ts

```diff
@@ -0,0 +1,9 @@
+declare global {
+    export type MultiReturn<T extends any[]> = T & {
+        readonly " __multiBrand": unique symbol;
+    };
+
+    export function multi<T extends any[]>(...values: T): MultiReturn<T>;
+}
+
+export {};
```

Comment:
There's a little reason to force file to be a module and use `declare global`
```suggestion
function multi<T extends any[]>(...values: T): MultiReturn<T>;

type MultiReturn<T extends any[]> = T & { readonly " __multiBrand": unique symbol; };
```